keyboards/sofle/keymaps/rgb_pev/keymap.c

```diff
@@ -0,0 +1,964 @@
+////        case _COLEMAK:
+//        case _QWERTY: //5 characters also creates a new line be default
+////        case _COLEMAKDH:
+//            oled_write_ln_P(PSTR("Base\n"), false);
+//            break;
+//
+//        case _ALT:
+//            oled_write_ln_P(PSTR("Alter"), false);
+//            break;
+//        case _NUM:
+//            oled_write_ln_P(PSTR("M-Num"), false);
+//            break;
+//        case _FN:
+//            oled_write_ln_P(PSTR("Func "), false);
+//            break;
+//        case _KB:
+//            oled_write_ln_P(PSTR("KBset"), false);
+//            break;
+//
+//        default:
+//            oled_write_ln_P(PSTR("Undef"), false);
+//    }
+//    if(g_led_state.caps_lock) {
+//        oled_write_P(PSTR("CapsL"), false);
+//    } else {
+//        oled_write_ln_P(PSTR(""), false);
+//    }
+//    if(g_led_state.num_lock) {
+//        oled_write_P(PSTR("NumL "), false);
+//    } else {
+//        oled_write_ln_P(PSTR(""), false);
+//    }
+//    if(g_led_state.scroll_lock) {
+//        oled_write_P(PSTR("ScrlL"), false);
+//    } else {
+//        oled_write_ln_P(PSTR(""), false);
+//    }
+//    oled_write_ln_P(PSTR(""), false);
+//
+//    char snum[5];
+////    int8_t snumLen;
+////    snumLen = ceil(log10(keyCntr));
+//    itoa(keyCntr%100000, snum, 10);
+////    if(snumLen < 5) {
+////        snum[snumLen+1] = '\n';
+////    }
+//    oled_write_ln(snum, false);
+////    oled_write_ln_P(snum, false);
+//
+////`    snumLen = ceil(log10(keyCntr)/log10(16));
+////`    itoa(keyCodeLast, snum, 16);
+////`    if(snumLen < 5) {
+////`        snum[snumLen+1] = '\n';
+////`    }
+////`    oled_write(snum, false);
+////`    oled_write_ln_P(PSTR(""), false);
+//}
+
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+    if (true) { //is_keyboard_master()) {
+        return OLED_ROTATION_270;
+    } 
+    return rotation;
+}
+
+bool oled_task_user(void) {
+    if (is_keyboard_master()) {
+        //leftOled();
+        oled_sugar();
+    } else {
+        oled_sugar();
+       // render_logo();
+    }
+    return false;
+}
+
+void user_sync_a_slave_handler(uint8_t in_buflen, const void* in_data, uint8_t out_buflen, void* out_data) {
+    keyCntr = *((const uint16_t *)in_data);
+}
+
+#endif
+
+
+void keyboard_post_init_user(void) {
+    // Enable the LED layers
+    rgblight_layers = my_rgb_layers;
+#ifdef OLED_ENABLE
+    transaction_register_rpc(USER_SYNC_A, user_sync_a_slave_handler);
+#endif
+	rgblight_mode(10);// haven't found a way to set this in a more useful way
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+
+    if(record->event.pressed) {
+#ifdef OLED_ENABLE
+        keyCntr++;
+//        transaction_rpc_exec(USER_SYNC_A, 0, &m2s, sizeof(s2m), &s2m)
+        transaction_rpc_send(USER_SYNC_A, sizeof(keyCntr), &keyCntr);
+#endif
+        keyCodeLast = keycode;
+    }
+    return true;
+}
+
+#ifdef ENCODER_ENABLE
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
```

Comment:
The only issue I have with encoder map is that I need to ignore every second activation of the encoder (unfortunately). Because for every indentation it of the encoder activates the keyboard twice.

I don't think I can do this with the encoder map

Reply:
Most likely you just need to add this to your `config.h`:
```c
#undef ENCODER_RESOLUTION
#define ENCODER_RESOLUTION 4
```
(The QMK default for `ENCODER_RESOLUTION` is 4, but the `sofle/rev1` keyboard sets it to 2, which probably matches the encoders that the original developer used.)

Reply:
This worked great thanks!